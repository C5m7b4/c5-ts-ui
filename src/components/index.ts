import {
  Button,
  NormalButtonType,
  NormalButtonSize,
  ButtonProps,
} from './Button';

export { Button, NormalButtonType, NormalButtonSize, ButtonProps };
