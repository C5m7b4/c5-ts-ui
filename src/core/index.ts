import Backdrop, { BackdropProps } from './Backdrop';
import Modal, { ModalProps } from './Modal';
import ModalWrapper, { ModalWrapperProps } from './ModalWrapper';

export {
  Backdrop,
  BackdropProps,
  Modal,
  ModalProps,
  ModalWrapper,
  ModalWrapperProps,
};
