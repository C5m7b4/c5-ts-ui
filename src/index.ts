import { Add } from './math';

export { Add };
