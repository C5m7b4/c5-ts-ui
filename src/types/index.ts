export type Theme = 'light' | 'dark' | 'colored';

export type Parsers =
  | 'babel'
  | 'css'
  | 'json'
  | 'flow'
  | 'babel-flow'
  | 'babel-ts'
  | 'typescript'
  | 'json'
  | 'markdown'
  | 'html';
