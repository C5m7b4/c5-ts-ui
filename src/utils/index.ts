export * from './colors';
export * from './reserved';
export * from './fns';
