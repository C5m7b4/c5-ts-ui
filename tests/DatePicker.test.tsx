import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import { DatePicker } from '../src';

describe('DatePicker tests', () => {
  const testfn = jest.fn();

  test('should render correctly', () => {
    const { container } = render(
      <DatePicker onChange={testfn} date={new Date('3/2/2023')} />
    );
    expect(container).toMatchSnapshot();
  });

  test('should handle month change backwards', () => {
    jest.useFakeTimers();
    const { container } = render(
      <DatePicker onChange={testfn} date={new Date('3/2/2023')} />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    jest.advanceTimersByTime(200);
    const btn = container.querySelector(
      '.mdp-head-button-left-arrow'
    ) as HTMLDivElement;
    fireEvent.click(btn);
    jest.advanceTimersByTime(200);
    const title = container.querySelector('#c5-dp-1-title') as HTMLDivElement;
    expect(title.innerHTML).toEqual('2/2/2023');
  });

  test('should handle month change forwards', () => {
    jest.useFakeTimers();
    const { container } = render(
      <DatePicker onChange={testfn} date={new Date('3/2/2023')} />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    jest.advanceTimersByTime(200);
    const btn = container.querySelector(
      '.mdp-head-button-right-arrow'
    ) as HTMLDivElement;
    fireEvent.click(btn);
    jest.advanceTimersByTime(200);
    const title = container.querySelector('#c5-dp-1-title') as HTMLDivElement;
    expect(title.innerHTML).toEqual('4/2/2023');
  });

  test('should render with an id', () => {
    const { container } = render(
      <DatePicker onChange={testfn} id={'1'} date={new Date('3/2/2023')} />
    );
    const input = container.querySelector(
      '#c5-dp-1 > input'
    ) as HTMLInputElement;
    expect(input.value).toEqual('2023-03-02');
  });

  test('should handle selecting a new date', () => {
    jest.useFakeTimers();
    const { container, getByText } = render(
      <DatePicker onChange={testfn} date={new Date('3/2/2023')} id="1" />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    jest.advanceTimersByTime(200);
    const day = getByText('15');
    fireEvent.click(day);
    jest.advanceTimersByTime(500);
    const input = container.querySelector(
      '#c5-dp-1 > input'
    ) as HTMLInputElement;
    expect(input.value).toEqual('2023-03-15');
  });

  test('should handle no default date', () => {
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    render(<DatePicker onChange={testfn} />);
  });

  test('should handle next month change when its a twelve', () => {
    jest.useFakeTimers();
    const { container } = render(
      <DatePicker onChange={testfn} date={new Date('12/2/2022')} />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    jest.advanceTimersByTime(200);
    const btn = container.querySelector(
      '.mdp-head-button-right-arrow'
    ) as HTMLDivElement;
    fireEvent.click(btn);
    jest.advanceTimersByTime(200);
    const title = container.querySelector('#c5-dp-1-title') as HTMLDivElement;
    expect(title.innerHTML).toEqual('1/2/2023');
  });

  test('should handle previous month change when its a one', () => {
    jest.useFakeTimers();
    const { container } = render(
      <DatePicker onChange={testfn} date={new Date('1/1/2023')} />
    );
    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    jest.advanceTimersByTime(200);
    const btn = container.querySelector(
      '.mdp-head-button-left-arrow'
    ) as HTMLDivElement;
    fireEvent.click(btn);
    jest.advanceTimersByTime(200);
    const title = container.querySelector('#c5-dp-1-title') as HTMLDivElement;
    expect(title.innerHTML).toEqual('12/1/2022');
  });

  test('should handle clicking outside', () => {
    jest.useFakeTimers();
    const { container } = render(
      <div>
        <p>Here is the first div</p>
        <button>button</button>
        <DatePicker onChange={testfn} date={new Date('1/1/2023')} />
      </div>
    );

    const div = container.querySelector('#c5-dp-1') as HTMLDivElement;
    fireEvent.click(div);
    jest.advanceTimersByTime(500);

    fireEvent.click(div);
    jest.advanceTimersByTime(500);

    // now we click on the backdrop
    const backdrop = container.querySelector(
      '#c5-dp-1-backdrop'
    ) as HTMLDivElement;
    fireEvent.click(backdrop);

    const input = container.querySelector(
      '#c5-dp-1 > input'
    ) as HTMLInputElement;
    expect(input.value).toEqual('2023-01-01');
  });
});
