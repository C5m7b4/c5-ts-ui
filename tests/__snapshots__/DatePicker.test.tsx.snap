// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker tests should render correctly 1`] = `
<div>
  <div
    class="c5-date-picker"
    id="c5-date-picker"
  >
    <div
      class="c5-dp-input"
      id="c5-dp-1"
    >
      <input
        type="date"
      />
    </div>
    <div
      id="c5-dp-1-backdrop"
      style="position: fixed; z-index: 1; top: 0px; bottom: 0px; left: 0px; right: 0px; display: none; cursor: pointer;"
    />
    <div
      class="color-picker-modal-wrapper"
      style="position: fixed; top: 50%; left: 50%; z-index: 2; transform: translate(-50%, -50%); transform-origin: left top; max-width: 100%; height: auto; display: none; animation: color-picker-fade-in .5s ease;"
    >
      <div
        class="mdp-container"
        style="position: absolute; top: 0px; left: 0px;"
      >
        <div
          class="mdp-content-container"
          id="c5-dp-1"
        >
          <div
            class="header row flex-middle"
          >
            <div
              class="mdp-button"
            >
              <div
                class="mdp-head-button-inner"
              >
                <span
                  class="mdp-head-button-left-arrow"
                />
              </div>
            </div>
            <div
              class="col col-center"
            >
              <span
                id="c5-dp-1-title"
              >
                3/2/2023
              </span>
            </div>
            <div
              class="mdp-button"
            >
              <div
                class="mdp-head-button-inner"
              >
                <span
                  class="mdp-head-button-right-arrow"
                />
              </div>
            </div>
          </div>
          <div
            class="days row"
          >
            <div
              class="cch-name"
            >
              S
            </div>
            <div
              class="cch-name"
            >
              M
            </div>
            <div
              class="cch-name"
            >
              T
            </div>
            <div
              class="cch-name"
            >
              W
            </div>
            <div
              class="cch-name"
            >
              T
            </div>
            <div
              class="cch-name"
            >
              F
            </div>
            <div
              class="cch-name"
            >
              S
            </div>
          </div>
          <div
            class="mpd-body"
          >
            <div
              class="row"
            >
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  26
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  27
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  28
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  1
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  2
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  3
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day highlighted-text"
                >
                  4
                </span>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  5
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  6
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  7
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  8
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  9
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  10
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  11
                </span>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  12
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  13
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  14
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  15
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  16
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  17
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  18
                </span>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  19
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  20
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  21
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  22
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  23
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  24
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  25
                </span>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  26
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  27
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  28
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  29
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  30
                </span>
              </div>
              <div
                class="c-day-container "
              >
                <span
                  class="cdc-day "
                >
                  31
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  1
                </span>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  2
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  3
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  4
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  5
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  6
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  7
                </span>
              </div>
              <div
                class="c-day-container disabled"
              >
                <span
                  class="cdc-day "
                >
                  8
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
